datajs/src/file-base.js

```diff
@@ -144,46 +201,12 @@ export class FileInterface extends File {
     return this.descriptor.name
   }
 
-
-
-
   /**
-   *
-   * @param {string} hashType - Should be md5 or sha256 
-   * @param {string} cbProgress - Should be a callback to track the progress
-   */
-  async generateHash(hashType, cbProgress) {
-    return new Promise((resolve, reject) => {
-      let newHash =  hashType === "md5" ? crypto.createHash('md5') : crypto.createHash('sha256');
-      readChunk(this.descriptor, (chunk, offs, total) => {
-        newHash.update(chunk);
-        if (cbProgress) {
-          cbProgress(offs / total);
-        }
-      }, err => {
-        if (err) {
-          reject(err);
-        } else {
-          let hashHex = newHash.digest('hex')
-          resolve(hashHex);
-        }
-      });
-    });
-  }
-
-  /**
-   *
-   * @param {string} cbProgress - Should be a callback to track the progress
-   */
-  async hash(cbProgress) {
-    return this.generateHash("md5", cbProgress)
-  }
-
-  /**
-   *
-   * @param {string} cbProgress - Should be a callback to track the progress
+   * Calculates the hash of a file
+   * @param {string} hashType - md5/sha256 type of hash algorithm to use
    */
-   async hashSha256(cbProgress) {
-    return this.generateHash("sha256", cbProgress)
+  async hash(hashType = 'sha256') {
+    let stream = webToNodeStream(this.descriptor.stream())
+    return computeHash(stream, this.size, hashType)
   }
```

Comment:
@risenW not sure if it's a good idea to change the hash method. For old versions, the method `hash()` generates by default MD5 hash. Maybe these changes can impact other projects if they update to this version and currently impact datapub because we a calling `hashSha256`:

![Screenshot from 2020-10-30 10-38-14](https://user-images.githubusercontent.com/24671133/97711603-0e2d4400-1a9c-11eb-87a7-608b9966b12b.png)

cc: @kmanaseryan

Reply:
@mariorodeghiero agree. @risenW we shouldn't close this PR without review. We should revert master branch

Reply:
@kmanaseryan I can revert this back to the old method. I wanted to unify the hash method into a single function.